arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -83,6 +83,24 @@ size_t RestCursorHandler::queue() const {
   return JobQueue::STANDARD_QUEUE;
 }
 
+/// @brief whether the request should be forwarded to a different server
+bool RestCursorHandler::shouldForwardRequest() {
```

Comment:
Does it make sense to combine shouldForwardRequest() and forwardingTarget()?
forwardingTarget() has a precondition that shouldForwardRequest() returns true, and will then carry out the same steps to determine the target. However, the target is already determined by shouldForwardRequest. It's simply not returned from it.
Maybe we can change the return type of shouldForwardRequest so that it returns the target as well?
I think if it's implemented in two different functions, it's easy to make changes to one of them and forget the other.